### PR TITLE
Describe test cases in a table

### PR DESCRIPTION
**Pull request by lukesiler**
@0xHiteshPatel 

#### What issues does this address?
Fixes #<issueid>
WIP #<issueid>
...

#### What's this change do?
This change adds a README.rst to tests folder.  This provides a browser rendering of table of test cases where we can summarize the goal of the test case and verification criteria.  As we expand test cases to cover iWF and add success criteria those can be highlighted here.

The initial focus is on knowledge transfer - Hitesh and I will be working through the table and recording summarization.  More PR's to come with that content.

#### Where should the reviewer start?
Check out https://github.com/F5Networks/f5-application-services-integration-iApp/tree/describe-test-cases/test to see rendering.

#### Any background context?
The goal here is to capture the context of these test cases and help engineering team ramp-up understanding of what each test case is configuring the BIG-IP to do.

### CODE REVIEW

File: test/README.rst [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+.. _test_vs_standard_udp.json: test_vs_standard_udp.json
+.. _test_vs_standard_udp_afm.json: test_vs_standard_udp_afm.json
+
+
+iWorkflow
+
++---------------------------------------+----------------------------------+---------------------------------------------------------+
+| Test Case                             | Success Criteria                 | Summary                                                 |
++=======================================+==================================+=========================================================+
+| coming soon                           |                                  |                                                         |
```
Comment (RafalKorepta) [inline test/README.rst]: Is this place good for write 7 use cases that are in [iWorkflow_json_payloads_v2.0.004.zip|https://github.com/F5Networks/f5-application-services-integration-iApp/releases/download/v2.0.004/iWorkflow_json_payloads_v2.0.004.zip]?

Or is the goal to combine BIG-IP tests with iWrokflow?
@KrystianMarek?

File: test/README.rst [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+.. _test_vs_standard_tcp_afm.json: test_vs_standard_tcp_afm.json
+.. _test_vs_standard_tcp_options.json: test_vs_standard_tcp_options.json
+.. _test_vs_standard_tcp_rd_auto.json: test_vs_standard_tcp_rd_auto.json
+.. _test_vs_standard_tcp_rd_nonauto.json: test_vs_standard_tcp_rd_nonauto.json
+.. _test_vs_standard_tcp_routeadv_all.json: test_vs_standard_tcp_routeadv_all.json
+.. _test_vs_standard_tcp_routeadv_always.json: test_vs_standard_tcp_routeadv_always.json
+.. _test_vs_standard_tcp_routeadv_any.json: test_vs_standard_tcp_routeadv_any.json
+.. _test_vs_standard_tcp_virt_addr_options.json: test_vs_standard_tcp_virt_addr_options.json
+.. _test_vs_standard_udp.json: test_vs_standard_udp.json
+.. _test_vs_standard_udp_afm.json: test_vs_standard_udp_afm.json
```
Comment (RafalKorepta) [inline test/README.rst]: The new test was merged #47 test_without_vs__Name.json

Comment (lukesiler) [inline test/README.rst]: ok, thx

File: test/README.rst (hunk 1 above)
Comment (lukesiler) [inline test/README.rst]: Each of the 7 iWF Service Templates will have 1 or more Service definitions.  These 7+ Service definitions will represent test cases that get listed here in the future.

I'm not sure about the tooling that @KrystianMarek has in mind or the file layout for these new tests.  Once that is known then we can figure out how to map those into this documentation.

**Pull request merged**